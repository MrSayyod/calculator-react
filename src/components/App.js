import Display from "./Display"
import ButtonPanel from "./ButtonPanel"

const App = function() {
  return (
    <React.Fragment>
      <Display />
      <ButtonPanel />
    </React.Fragment>
  )
}

export default App