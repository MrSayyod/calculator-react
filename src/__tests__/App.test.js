import React from 'react';
// import renderer from "react-test-renderer";
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';
import App from '../components/App';

let container = null;
beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('renders correctly', () => {
  act(() => {
    render(<App />, container);
  });
  expect(
    pretty(container.innerHTML),
  )
    .toMatchSnapshot();
});
