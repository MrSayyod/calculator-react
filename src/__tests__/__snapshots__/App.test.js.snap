// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
"<header>
  <nav>
    <div class=\\"leftNav\\" data-testid=\\"navbar-header-test\\">Math Magicians</div>
    <div class=\\"rightNav\\" data-testid=\\"links-test\\">
      <ul>
        <li><a href=\\"/\\">Home</a></li>
        <li><a href=\\"/calculator\\">Calculator</a></li>
        <li><a href=\\"/quotes\\">Quotes</a></li>
      </ul>
    </div>
  </nav>
</header>
<div class=\\"displayResult\\">
  <div class=\\"display\\" data-testid=\\"display-test\\">0 </div>
  <div class=\\"btnContainer\\">
    <div class=\\"group_1\\"><button style=\\"background: orange;\\" type=\\"button\\" class=\\"otherBtn btn btn-primary\\">AC</button><button style=\\"background: orange;\\" type=\\"button\\" class=\\"otherBtn btn btn-primary\\">+/-</button><button style=\\"background: orange;\\" type=\\"button\\" class=\\"otherBtn btn btn-primary\\">%</button><button style=\\"background: orange;\\" type=\\"button\\" class=\\"otherBtn btn btn-primary\\">/</button></div>
    <div class=\\"group_2\\"><button style=\\"background: rgb(239, 239, 239);\\" type=\\"button\\" class=\\"otherBtn btn btn-primary\\">7</button><button style=\\"background: rgb(239, 239, 239);\\" type=\\"button\\" class=\\"otherBtn btn btn-primary\\">8</button><button style=\\"background: rgb(239, 239, 239);\\" type=\\"button\\" class=\\"otherBtn btn btn-primary\\">9</button><button style=\\"background: orange;\\" type=\\"button\\" class=\\"otherBtn btn btn-primary\\">*</button></div>
    <div class=\\"group_3\\"><button style=\\"background: rgb(239, 239, 239);\\" type=\\"button\\" class=\\"otherBtn btn btn-primary\\">4</button><button style=\\"background: rgb(239, 239, 239);\\" type=\\"button\\" class=\\"otherBtn btn btn-primary\\">5</button><button style=\\"background: rgb(239, 239, 239);\\" type=\\"button\\" class=\\"otherBtn btn btn-primary\\">6</button><button style=\\"background: orange;\\" type=\\"button\\" class=\\"otherBtn btn btn-primary\\">-</button></div>
    <div class=\\"group_4\\"><button style=\\"background: rgb(239, 239, 239);\\" type=\\"button\\" class=\\"otherBtn btn btn-primary\\">1</button><button style=\\"background: rgb(239, 239, 239);\\" type=\\"button\\" class=\\"otherBtn btn btn-primary\\">2</button><button style=\\"background: rgb(239, 239, 239);\\" type=\\"button\\" class=\\"otherBtn btn btn-primary\\">3</button><button style=\\"background: orange;\\" type=\\"button\\" class=\\"otherBtn btn btn-primary\\">+</button></div>
    <div class=\\"group_5\\"><button style=\\"background: rgb(239, 239, 239);\\" type=\\"button\\" class=\\"zeroBtn btn btn-primary\\">0</button><button style=\\"background: rgb(239, 239, 239);\\" type=\\"button\\" class=\\"otherBtn btn btn-primary\\">.</button><button style=\\"background: orange;\\" type=\\"button\\" class=\\"otherBtn btn btn-primary\\">=</button></div>
  </div>
</div>"
`;
